docs/workflow/trimming/feature-switches.md

```diff
@@ -8,18 +8,20 @@ configurations but their defaults might vary as any SDK can set the defaults dif
 | DebuggerSupport | System.Diagnostics.Debugger.IsSupported | Any dependency that enables better debugging experience to be trimmed when set to false |
+| EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization | System.ComponentModel.TypeConverter.EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization | BinaryFormatter serialization support is trimmed when set to false. |
 | EnableUnsafeUTF7Encoding | System.Text.Encoding.EnableUnsafeUTF7Encoding | Insecure UTF-7 encoding is trimmed when set to false |
 | EnableUnsafeBinaryFormatterSerialization | System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization | BinaryFormatter serialization support is trimmed when set to false |
 | EventSourceSupport | System.Diagnostics.Tracing.EventSource.IsSupported | Any EventSource related code or logic is trimmed when set to false |
 | InvariantGlobalization | System.Globalization.Invariant | All globalization specific code and data is trimmed when set to true |
-| UseSystemResourceKeys | System.Resources.UseSystemResourceKeys |  Any localizable resources for system assemblies is trimmed when set to true |
 | HttpActivityPropagationSupport | System.Net.Http.EnableActivityPropagation | Any dependency related to diagnostics support for System.Net.Http is trimmed when set to false |
 | UseNativeHttpHandler | System.Net.Http.UseNativeHttpHandler | HttpClient uses by default platform native implementation of HttpMessageHandler if set to true. |
 | StartupHookSupport | System.StartupHookProvider.IsSupported | Startup hooks are disabled when set to false. Startup hook related functionality can be trimmed. |
+| UseSystemResourceKeys | System.Resources.UseSystemResourceKeys |  Any localizable resources for system assemblies is trimmed when set to true |
 | TBD | System.Threading.ThreadPool.EnableDispatchAutoreleasePool | When set to true, creates an NSAutoreleasePool around each thread pool work item on applicable platforms. |
-| CustomResourceTypesSupport | System.Resources.ResourceManager.AllowCustomResourceTypes | Use of custom resource types is disabled when set to false. ResourceManager code paths that use reflection for custom types can be trimmed. |
-| EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization | System.ComponentModel.TypeConverter.EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization | BinaryFormatter serialization support is trimmed when set to false. |
+| _EnableCallingManagedFunctionFromNative | Internal.Runtime.InteropServices.ComponentActivator.IsSupported | Getting a managed function from native hosting is disabled when set to false and related functionality can be trimmed. |
+| _EnableIJWHostActivation | Internal.Runtime.InteropServices.InMemoryAssemblyLoader.IsSupported | Enabling an IJW host to load managed code is disabled when set to false and related functionality can be trimmed. |
```

Comment:
Is `IJW host` a common term/acronym? I'm not sure I've ever seen that term before, so I wonder if someone reading this would know what it means.

Reply:
This should use the "C++/CLI assembly" terminology.

Reply:
Is the C++/CLI switch internal (prefixed with the underscore) because we're considering it to be this special like the native hosting?
This seems like it would be more of a normal case to me - self-contained app using a C++/CLI assembly. I know we have https://github.com/dotnet/sdk/issues/12345 for publishing not working properly right now - it is marked for 6.0.1xx, so should be a .NET 6 scenario.

Reply:
@elinor-fung I never remember how this works... if I load a C++/CLI assembly from managed code (via `Assembly.Load` or just plain assembly ref) - does it go through the hosting layers?

Reply:
`Assembly.Load` or plain assembly ref won't go through the hosting layers. P/Invoke into the assembly (which we've gotten customer repro projects doing) will go through the hosting layers.

Reply:
> P/Invoke into the assembly (which we've gotten customer repro projects doing) will go through the hosting layers.

In that case I think we should make this a public MSBuild property. We don't have to advertise it, it's more about being prepared when somebody hits this. I would still keep it off by default - we just need to make sure there's a reasonable error experience.

Reply:
The p/invoke case sounds pretty hairy for illink.

If I have a C++/CLI assembly that is only referenced through a p/invoke from the main app, illink should probably consider the C++/CLI assembly `copyused` and scan it, even though it doesn't show up as one of the input assemblies.

Do we need a illink issue to track this? How common is that?